elements/lisk-chain/src/state_store/state_store.ts

```diff
@@ -56,11 +57,9 @@ export class StateStore {
 		this.chain.restoreSnapshot();
 	}
 
-	public async finalize(tx: StorageTransaction): Promise<void> {
-		await Promise.all([
-			this.account.finalize(tx),
-			this.chain.finalize(tx),
-			this.consensus.finalize(tx),
-		]);
+	public finalize(batch: BatchChain): void {
```

Comment:
Seems previous code was using a SQL transaction how does this change guaranteed that either all or none of the updates happen as to maintain consistency?

Reply:
As mentioned above, `batch.write` will save to the database with consistency.